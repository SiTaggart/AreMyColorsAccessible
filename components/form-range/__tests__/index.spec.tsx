/* eslint-env jest */
/// <reference types="jest" />

import React, { ReactElement } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { FormRange, FormRangeProps } from '..';

type IFormRangeWrapperProps = Partial<FormRangeProps>;

describe('FormRange', (): void => {
  const FormRangeWrapper: React.FunctionComponent<IFormRangeWrapperProps> = (
    props: IFormRangeWrapperProps
  ): ReactElement<HTMLInputElement> => <FormRange id="range-id" max={100} min={1} {...props} />;

  it('renders without crashing', (): void => {
    ReactDOM.render(<FormRangeWrapper />, document.createElement('div'));
    const formRangeCmp = renderer.create(<FormRangeWrapper />).toJSON();
    expect(formRangeCmp).toMatchSnapshot();
  });

  it('should set a default value', (): void => {
    const { getByTestId } = render(
      <FormRangeWrapper defaultValue="testing" id="default-value-test" />
    );
    expect(getByTestId('default-value-test').getAttribute('value')).toEqual('testing');
  });

  it('should call a passed in onChange method, onChange', (): void => {
    const onChangeMock: jest.Mock = jest.fn();
    const { getByTestId } = render(<FormRangeWrapper id="onchange-test" onChange={onChangeMock} />);
    fireEvent.change(getByTestId('onchange-test'), { target: { value: 23 } });
    expect(onChangeMock).toHaveBeenCalled();
  });

  it('should call a passed in onInput method, onInput', (): void => {
    const onInputMock: jest.Mock = jest.fn();
    const { getByTestId } = render(<FormRangeWrapper id="oninput-test" onInput={onInputMock} />);
    fireEvent.input(getByTestId('oninput-test'), { target: { value: 12 } });
    expect(onInputMock).toHaveBeenCalled();
  });

  it('should set the value of the input when passed a value', (): void => {
    const onChangeMock: jest.Mock = jest.fn();
    const { getByTestId } = render(
      <FormRangeWrapper id="value-test" onChange={onChangeMock} value={3} />
    );
    expect(getByTestId('value-test').getAttribute('value')).toEqual('3');
  });
});
