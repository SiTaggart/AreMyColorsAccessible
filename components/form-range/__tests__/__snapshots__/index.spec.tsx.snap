// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRange renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-0::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-0::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-0:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-0::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-0::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-0::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-0::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-0::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-0::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-0:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-0:focus::-ms-fill-upper {
  background: currentColor;
}

<input
    class="emotion-0 emotion-1"
    data-testid="range-id"
    id="range-id"
    max="100"
    min="1"
    type="range"
    value=""
  />
</DocumentFragment>
`;
